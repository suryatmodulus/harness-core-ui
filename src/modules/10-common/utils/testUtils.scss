/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
