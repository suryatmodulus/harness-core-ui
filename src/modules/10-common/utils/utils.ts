/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
