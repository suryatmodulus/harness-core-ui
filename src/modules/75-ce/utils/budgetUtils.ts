/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum AlertStatus {
  Danger,
  Warning,
  Info
}

interface CostStatusInfo {
  cost: number
  ratio: number
  percentage: number
  status: AlertStatus
}

export interface getAllBudgetCostInfoReturnType {
  actualCostStatus: CostStatusInfo
  forecastedCostStatus: CostStatusInfo
  budgetAmountStatus: CostStatusInfo
}

export const getAllBudgetCostInfo: (
  forecastedCost: number,
  actualCost: number,
  budgetAmount: number
) => getAllBudgetCostInfoReturnType = (forecastedCost, actualCost, budgetAmount) => {
  /**
   * Find out the max cost of all and it will be considered as 100%
   * and the rest of the cost percentages will be calculated based on that
   */

  const baseCost = Math.max(actualCost, forecastedCost, budgetAmount)
  const actualCostRatio = actualCost / baseCost
  const forecastCostRatio = forecastedCost / baseCost
  const budgetAmountRatio = budgetAmount / baseCost

  const actualCostPerWithBudget = (actualCost / budgetAmount) * 100
  const forecastCostPerWithBudget = (forecastedCost / budgetAmount) * 100

  const getAlertStatus: (percentage: number) => AlertStatus = percentage => {
    if (percentage > 100) {
      return AlertStatus.Danger
    }
    if (percentage > 90) {
      return AlertStatus.Warning
    }
    return AlertStatus.Info
  }

  const actualCostStatus: CostStatusInfo = {
    cost: actualCost,
    ratio: actualCostRatio,
    percentage: actualCostPerWithBudget,
    status: getAlertStatus(actualCostPerWithBudget)
  }
  const forecastedCostStatus: CostStatusInfo = {
    cost: forecastedCost,
    ratio: forecastCostRatio,
    percentage: forecastCostPerWithBudget,
    status: getAlertStatus(forecastCostPerWithBudget)
  }
  const budgetAmountStatus: CostStatusInfo = {
    cost: budgetAmount,
    ratio: budgetAmountRatio,
    percentage: 100,
    status: AlertStatus.Info
  }

  return {
    actualCostStatus,
    forecastedCostStatus,
    budgetAmountStatus
  }
}
