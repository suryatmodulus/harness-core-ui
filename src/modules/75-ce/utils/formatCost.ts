/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

interface FormatCostOptions {
  currency?: string
  locale?: string
  decimalPoints?: number
  shortFormat?: boolean
}

type FormatCost = (value: number, options?: FormatCostOptions) => string

const defaultOptions = {
  shortFormat: false,
  currency: 'USD',
  locale: 'en-us',
  decimalPoints: 2
}

const TOTAL_DIGITS_TO_SHOW = 3
const POWER_OF_TEN_IN_BILLION = 9
const POWER_OF_TEN_IN_MILLION = 6
const POWER_OF_TEN_IN_THOUSANDS = 3

type convertNumberToLocaleStringType = (
  value: number,
  locale: string,
  currency: string,
  maximumFractionDigits: number,
  minimumFractionDigits: number
) => string

const convertNumberToLocaleString: convertNumberToLocaleStringType = (
  value,
  locale,
  currency,
  maximumFractionDigits,
  minimumFractionDigits
) =>
  value.toLocaleString(locale, {
    style: 'currency',
    currency,
    maximumFractionDigits,
    minimumFractionDigits
  })

const formatCost: FormatCost = (value: number, options?: FormatCostOptions) => {
  const costOptions = { ...defaultOptions, ...options }
  if (isNaN(value)) {
    return ''
  }

  const getCostInShortFormat = (power: number, cost: number, suffix: string, numberOfDigits: number) => {
    const decimalPlacesToHave = Math.max(power + TOTAL_DIGITS_TO_SHOW - numberOfDigits, 0)
    return (
      convertNumberToLocaleString(
        cost / Math.pow(10, power),
        costOptions.locale,
        costOptions.currency,
        decimalPlacesToHave,
        decimalPlacesToHave
      ) + suffix
    )
  }

  /**
    Option shortFormat will convert the cost as mentioned below -
    2,345,456 as $2.34M
    12,345,456 as $12.3M
    123,456,789 as $123M
    and so on for thousands and billions
  */
  if (costOptions.shortFormat) {
    const numberOfDigits = value.toFixed(0).length
    if (numberOfDigits > POWER_OF_TEN_IN_BILLION) {
      return getCostInShortFormat(POWER_OF_TEN_IN_BILLION, value, 'B', numberOfDigits)
    }

    if (numberOfDigits > 6) {
      return getCostInShortFormat(POWER_OF_TEN_IN_MILLION, value, 'M', numberOfDigits)
    }

    if (numberOfDigits > 3) {
      return getCostInShortFormat(POWER_OF_TEN_IN_THOUSANDS, value, 'K', numberOfDigits)
    }
  }

  return convertNumberToLocaleString(
    value,
    costOptions.locale,
    costOptions.currency,
    costOptions.decimalPoints,
    costOptions.decimalPoints
  )
}

export default formatCost
