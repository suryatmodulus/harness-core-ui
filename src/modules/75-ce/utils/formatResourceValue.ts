/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

const BYTES_IN_A_GB = 1000000000
const BYTES_IN_A_MB = 1000000

type fnNumberToString = (val: number) => string

const log10: (n: number) => number = n => Math.log(n) / Math.log(10)

export const getCPUValueInReadableForm: (val: number) => string = (val: number) => {
  if (!val) {
    return '0'
  }
  if (log10(val) < 0) {
    return `${Math.ceil(val * 1000)}m`
  }
  return val % 1 === 0 ? `${val}` : `${(val + 0.1).toFixed(1)}`
}

export const getMemValueInReadableForm: (val: number) => string = (val: number) => {
  if (!val) {
    return '0'
  }
  if (log10(val) >= 9) {
    return val % 1 === 0 ? `${val / BYTES_IN_A_GB}Gi` : `${(val / BYTES_IN_A_GB + 0.1).toFixed(1)}Gi`
  } else {
    return `${Math.ceil(val / BYTES_IN_A_MB)}Mi`
  }
}

export const getMemValueInReadableFormForChart: fnNumberToString = val => {
  if (!val) {
    return '0'
  }
  if (log10(val) >= 9) {
    return `${(val / BYTES_IN_A_GB).toFixed(2)}Gi`
  } else {
    return `${(val / BYTES_IN_A_MB).toFixed(2)}Mi`
  }
}

export const getCPUValueInmCPUs: fnNumberToString = val => {
  return `${val * 1000}`
}

export const getMemValueInGB: fnNumberToString = val => {
  return `${val / BYTES_IN_A_GB}Gi`
}

export const getRecommendationYaml: (
  cpuReqVal: string | number,
  memReqVal: string | number,
  memLimitVal: string | number
) => string = (cpuReqVal, memReqVal, memLimitVal) => {
  return `limits:\n  memory: ${memLimitVal}\nrequests:\n  memory: ${memReqVal}\n  cpu: ${cpuReqVal}\n`
}

export const getMemoryValueInGBFromExpression: (val: string | number) => number = val => {
  const stringVal = `${val}`
  if (stringVal.includes('Gi')) {
    return Number(stringVal.split('Gi')[0])
  } else if (stringVal.includes('Mi')) {
    return Number(stringVal.split('Mi')[0]) / 1000
  }
  return Number(val) / BYTES_IN_A_GB
}

export const getCPUValueInCPUFromExpression: (val: string | number) => number = val => {
  const stringVal = `${val}`
  if (stringVal.includes('m')) {
    return Number(stringVal.split('m')[0]) / 1000
  }
  return Number(val)
}
