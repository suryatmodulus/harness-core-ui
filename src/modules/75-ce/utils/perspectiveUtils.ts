/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@wings-software/uicore'
import type Highcharts from 'highcharts'
import type { ViewRule, ViewIdCondition } from 'services/ce'
import {
  QlceViewTimeFilterOperator,
  ViewFieldIdentifier,
  QlceViewFilterWrapperInput,
  QlceViewFieldInputInput,
  QlceViewGroupByInput,
  QlceViewTimeGroupType,
  QlceViewFilterInput,
  ViewTimeRangeType,
  ViewChartType,
  QlceViewRuleInput
} from 'services/ce/services'
import { DATE_RANGE_SHORTCUTS } from './momentUtils'

const startTimeLabel = 'startTime'

export const getViewFilterForId: (viewId: string, isPreview?: boolean) => QlceViewFilterWrapperInput = (
  viewId: string,
  isPreview = false
) => {
  return { viewMetadataFilter: { viewId: viewId, isPreview: isPreview } } as QlceViewFilterWrapperInput
}

export const getTimeFilters: (from: number, to: number) => QlceViewFilterWrapperInput[] = (from, to) => {
  return [
    {
      timeFilter: {
        field: { fieldId: startTimeLabel, fieldName: startTimeLabel, identifier: ViewFieldIdentifier.Common },
        operator: QlceViewTimeFilterOperator.After,
        value: from
      }
    },
    {
      timeFilter: {
        field: { fieldId: startTimeLabel, fieldName: startTimeLabel, identifier: ViewFieldIdentifier.Common },
        operator: QlceViewTimeFilterOperator.Before,
        value: to
      }
    }
  ] as QlceViewFilterWrapperInput[]
}

export const getGroupByFilter: (groupBy: QlceViewFieldInputInput) => QlceViewGroupByInput = groupBy => {
  return {
    entityGroupBy: groupBy
  } as QlceViewGroupByInput
}

export const getTimeRangeFilter: (aggregation: QlceViewTimeGroupType) => QlceViewGroupByInput = aggregation => {
  return {
    timeTruncGroupBy: {
      resolution: aggregation
    }
  } as QlceViewGroupByInput
}

export const getFilters: (viewConditions: QlceViewFilterInput[]) => QlceViewFilterWrapperInput[] = viewConditions => {
  return viewConditions
    .filter(viewCondition => {
      return viewCondition.values?.length
    })
    .map(viewCondition => {
      return {
        idFilter: {
          values: viewCondition.values,
          operator: viewCondition.operator,
          field: {
            fieldId: viewCondition.field.fieldId,
            fieldName: viewCondition.field.fieldName,
            identifier: viewCondition.field.identifier
          }
        }
      }
    }) as QlceViewFilterWrapperInput[]
}

export const DEFAULT_GROUP_BY = {
  fieldId: 'product',
  fieldName: 'Product',
  identifier: ViewFieldIdentifier.Common,
  identifierName: ViewFieldIdentifier.Common
}

export const GROUP_BY_POD = {
  fieldId: 'instanceId',
  fieldName: 'Pod',
  identifier: ViewFieldIdentifier.Cluster,
  identifierName: ViewFieldIdentifier.Cluster
}

export const generateId: (length: number) => string = length => {
  let result = ''
  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
  const charactersLength = characters.length
  for (let i = 0; i < length; i++) {
    result += characters.charAt(Math.floor(Math.random() * charactersLength))
  }
  return result
}

export const CREATE_CALL_OBJECT = {
  viewVersion: 'v1',
  viewTimeRange: {
    viewTimeRangeType: ViewTimeRangeType.Last_7
  },
  viewType: 'CUSTOMER',
  viewVisualization: {
    granularity: QlceViewTimeGroupType.Day,
    groupBy: { ...DEFAULT_GROUP_BY },
    chartType: ViewChartType.StackedLineChart
  }
}

export const normalizeViewRules: (viewRules: ViewRule[] | undefined) => QlceViewRuleInput[] = viewRules => {
  return viewRules
    ? (viewRules
        .map(rule => {
          if (rule) {
            return {
              conditions:
                rule.viewConditions &&
                rule.viewConditions
                  .map(c => {
                    const condition = c as unknown as ViewIdCondition
                    return condition.values && condition.values.length
                      ? {
                          field: {
                            fieldId: condition.viewField?.fieldId,
                            fieldName: condition.viewField?.fieldName,
                            identifier: condition.viewField?.identifier,
                            identifierName: condition.viewField?.identifierName
                          },
                          operator: condition.viewOperator,
                          values: condition.values
                        }
                      : null
                  })
                  .filter(condition => !!condition)
            }
          }
        })
        .filter(rule => rule?.conditions && rule.conditions.length) as QlceViewRuleInput[])
    : []
}

export const getRuleFilters: (rules: QlceViewRuleInput[]) => QlceViewFilterWrapperInput[] = rules => {
  return rules.map(rule => ({
    ruleFilter: rule
  })) as QlceViewFilterWrapperInput[]
}

export const SOURCE_ICON_MAPPING: Record<string, IconName> = {
  AWS: 'service-aws',
  GCP: 'gcp',
  CLUSTER: 'blue-black-cluster',
  CUSTOM: 'pipeline-custom',
  AZURE: 'service-azure'
}

export const perspectiveDefaultTimeRangeMapper: Record<string, moment.Moment[]> = {
  [ViewTimeRangeType.Last_7]: DATE_RANGE_SHORTCUTS.LAST_7_DAYS,
  [ViewTimeRangeType.Last_30]: DATE_RANGE_SHORTCUTS.LAST_30_DAYS,
  [ViewTimeRangeType.LastMonth]: DATE_RANGE_SHORTCUTS.LAST_MONTH
}

export enum ChartState {
  IN_ACTIVE = 'inactive',
  NONE = ''
}

export const highlightNode = (chartRef: React.RefObject<Highcharts.Chart | undefined>, id: string) => {
  if (chartRef && chartRef.current) {
    const chart = chartRef.current
    chart.series?.length > 1 &&
      chart.series.forEach(chartItem => {
        const nodeId = (chartItem.options as any).nodeId
        if (nodeId !== id) {
          chartItem.setState(ChartState.IN_ACTIVE)
        }
      })
  }
}

export const resetNodeState = (chartRef: React.RefObject<Highcharts.Chart | undefined>) => {
  if (chartRef && chartRef.current) {
    const chart = chartRef.current
    chart.series &&
      chart.series.forEach(chartItem => {
        chartItem.setState(ChartState.NONE)
      })
  }
}

export const clusterInfoUtil: (dataSources?: string[]) => { isClusterOnly: boolean; hasClusterAsSource: boolean } =
  dataSources => {
    let isClusterOnly = false
    let hasClusterAsSource = false
    if (!dataSources?.length) {
      return { isClusterOnly, hasClusterAsSource }
    }
    if (dataSources.length === 1 && dataSources[0] === ViewFieldIdentifier.Cluster) {
      isClusterOnly = true
    }
    if (dataSources.includes(ViewFieldIdentifier.Cluster)) {
      hasClusterAsSource = true
    }
    return { isClusterOnly, hasClusterAsSource }
  }
