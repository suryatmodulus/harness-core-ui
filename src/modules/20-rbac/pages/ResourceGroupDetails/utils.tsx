import produce from 'immer'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector } from '@rbac/utils/utils'
import type { ResourceSelector } from 'services/resourcegroups'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (isDynamicResourceSelector(resourceSelector.type)) {
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    } else {
      map.set(resourceSelector.resourceType, resourceSelector.identifiers)
    }
  })
  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}
