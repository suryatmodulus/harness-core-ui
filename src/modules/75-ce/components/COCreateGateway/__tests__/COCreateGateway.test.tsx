/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import CECODashboardPage from '../COCreateGateway'

jest.mock('@common/components/YAMLBuilder/YamlBuilder')

jest.mock('services/cd-ng', () => ({
  useGetConnector: jest.fn().mockImplementation(() => ({ loading: false, refetch: jest.fn(), data: undefined }))
}))

describe('Create Auto Stopping rule', () => {
  test('Render cloud provider & connector selection screen', async () => {
    const { container } = render(
      <TestWrapper>
        <CECODashboardPage />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
