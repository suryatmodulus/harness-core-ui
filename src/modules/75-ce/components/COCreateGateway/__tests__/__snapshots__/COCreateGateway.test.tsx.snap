// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Auto Stopping rule Render cloud provider & connector selection screen 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--background-white"
    style="height: 100vh;"
  >
    <div
      class="StyledProps--font StyledProps--main StyledProps--background-white"
      style="height: 100vh;"
    >
      <div
        class="breadCrumb breadcrumbs"
      >
        <div>
          <a
            class="breadcrumb"
            href="/account/undefined/ce/autostopping-rules"
          >
            ce.co.breadCrumb.rules
          </a>
          <span
            class="bp3-icon StyledProps--main separator"
            data-icon="main-chevron-right"
          >
            <svg
              height="11"
              viewBox="0 0 8 14"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                fill="#4d5969"
              />
            </svg>
          </span>
        </div>
        <div>
          <span
            class="lastBreadcrumb"
          />
        </div>
      </div>
      <div
        style="top: 35px; box-shadow: rgb(40 41 61 / 4%) 0px 2px 8px, rgb(96 97 112 / 16%) 0px 16px 24px; width: 392px; bottom: 0px; right: 0px; position: fixed; z-index: 20; overflow-y: scroll;"
      >
        <div
          class="StyledProps--font StyledProps--main"
        >
          <div
            class="StyledProps--font StyledProps--main StyledProps--padding-large"
          >
            <div
              class="Layout--vertical Layout--layout-spacing-xxlarge StyledProps--main StyledProps--padding-medium"
            >
              <h2
                class="StyledProps--font StyledProps--font-h2 StyledProps--main"
                level="2"
              >
                Select Cloud Account
              </h2>
              <img
                alt=""
                aria-hidden="true"
                src="test-file-stub"
                style="width: 50%; align-self: center;"
              />
              <p
                class="StyledProps--font StyledProps--main"
              >
                ce.co.autoStoppingRule.providerSelector.helpText.selectProvider
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main"
        style="margin: 0px auto; padding-top: 100px; padding-left: 50px;"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-large StyledProps--main StyledProps--padding-large"
        >
          <h1
            class="StyledProps--font StyledProps--font-h1 StyledProps--main title"
            data-tooltip-id="coGetStarted"
          >
            common.letsGetYouStarted
          </h1>
          <h2
            class="StyledProps--font StyledProps--font-h2 StyledProps--main"
            level="2"
          >
            ce.co.autoStoppingRule.providerSelector.description
          </h2>
          <div
            class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
              style="padding-top: 29px;"
            >
              <div
                class="providersViewGrid"
              >
                <div
                  class="bp3-card bp3-interactive bp3-elevation-0 Card--card Card--interactive"
                  data-index="0"
                  tabindex="0"
                >
                  <div
                    class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                  >
                    <div
                      class="Card--cardIcon card"
                    >
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="service-aws"
                      >
                        <svg
                          height="28"
                          viewBox="0 0 14 10"
                          width="28"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4 3.9v.4c0 .1.1.2.1.3v.1c0 .1 0 .1-.1.1l-.1.2h-.1s-.1 0-.1-.1c-.1.1-.2 0-.2-.1s-.1-.2-.1-.3c-.4.5-.9.7-1.4.7-.4 0-.7-.1-.9-.3S.7 4.3.7 4c0-.4.1-.7.4-.9s.6-.4 1.1-.4h.5c.2 0 .3.1.5.1v-.3c0-.3-.1-.6-.2-.7-.1-.2-.3-.2-.7-.2-.2 0-.3 0-.5.1-.2 0-.3.1-.5.2 0 0-.1 0-.2.1H1v-.5-.2l.1-.1c.3-.1.4-.1.6-.2.3 0 .5-.1.8-.1.5 0 .9.1 1.2.4s.3.6.3 1.1zm-1.8.7c.2 0 .3 0 .5-.1.1-.1.3-.2.4-.3s.1-.2.1-.3V3.3c-.1 0-.2 0-.3-.1h-.4c-.3 0-.5.1-.7.2s-.3.3-.3.5.1.4.2.5.3.2.5.2zm3.6.5h-.2s-.1-.1-.1-.2l-1-3.6v-.2c0-.1 0-.1.1-.1H5.2s.1.1.1.2l.8 3 .7-3c0-.1 0-.1.1-.2h.8s.1.1.1.2l.7 3 .8-3c0-.1 0-.1.1-.2H10c.1 0 .1 0 .1.1v.2L8.8 4.8c0 .1 0 .2-.1.2h-.5c-.1.1-.2.1-.2 0 0 0-.1-.1-.1-.2l-.7-2.9-.7 2.9c0 .1 0 .1-.1.2h-.2zm5.9.1c-.2 0-.5 0-.7-.1s-.4-.1-.5-.2c-.1 0-.1-.1-.1-.1v-.4c0-.1 0-.1.1-.1h.2c.2.1.3.1.5.2.2 0 .4.1.5.1.3 0 .5 0 .7-.1.1-.1.2-.3.2-.4s0-.2-.1-.3-.2-.2-.5-.2l-.7-.2c-.3-.1-.6-.3-.7-.5s-.2-.4-.2-.7c0-.2 0-.3.1-.5.1-.1.2-.3.3-.4s.3-.2.5-.2c.2-.1.4-.1.6-.1h.6c.1 0 .2 0 .3.1.1 0 .1 0 .2.1.1 0 .1.1.1.1V1.6c0 .1 0 .1-.1.1s-.1 0-.2-.1c-.3-.1-.6-.2-.9-.2s-.5 0-.6.1-.2.2-.2.4c0 .1 0 .2.1.3s.3.2.5.2l.6.2c.3.1.6.2.7.4s.2.4.2.6 0 .4-.1.5c-.1.2-.2.3-.3.4-.1.5-.3.5-.5.6s-.4.1-.6.1z"
                            fill="#252f3e"
                          />
                          <g
                            fill="#f90"
                          >
                            <path
                              d="M12.5 7.4C11 8.5 8.9 9.1 7 9.1c-2.6 0-5-1-6.7-2.6-.1-.1 0-.3.2-.2 1.9 1.1 4.3 1.8 6.7 1.8 1.8 0 3.5-.4 5.1-1.1.2-.1.5.2.2.4z"
                            />
                            <path
                              d="M13.1 6.7c-.2-.2-1.3-.1-1.7-.1-.1 0-.2-.1 0-.2.9-.6 2.3-.4 2.4-.2.2.2 0 1.6-.8 2.3-.1.1-.2 0-.2-.1.1-.5.5-1.5.3-1.7z"
                            />
                          </g>
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="bp3-card bp3-interactive bp3-elevation-0 Card--card Card--interactive"
                  data-index="1"
                  tabindex="0"
                >
                  <div
                    class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                  >
                    <div
                      class="Card--cardIcon card"
                    >
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="service-azure"
                      >
                        <svg
                          height="28"
                          viewBox="0 0 14 12"
                          width="28"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="#2075b7"
                          >
                            <path
                              d="M14 11.3H3.5c2.1-.4 4.2-.7 6.3-1.1 0-.1 0-.1-.1-.1-1.1-1.4-2.2-2.7-3.2-3.9-.1-.1-.1-.2 0-.2.4-1.1.8-2.1 1.1-3.2.2-.5.4-1 .5-1.5 0 0 0-.1.1-.1C10.1 4.6 12 7.9 14 11.3zM7.7.5c-.2.4-.4.7-.5 1.1-1 2.1-1.9 4.1-2.9 6.2-.3.7-.7 1.4-1 2.2 0 .1-.1.1-.2.1H.2c-.2 0-.2 0-.1-.1 1.1-2 2.3-3.9 3.4-5.9 0-.1.1-.2.2-.3C5 2.8 6.3 1.7 7.6.6c0 0 0-.1.1-.1z"
                            />
                          </g>
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main instanceTypeNameGrid"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-align-center"
                style="font-size: 11px;"
              >
                AWS
              </p>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-align-center"
                style="font-size: 11px;"
              >
                Azure
              </p>
            </div>
          </div>
        </div>
        <button
          class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main footer StyledProps--intent-primary Button--with-current-color Button--withLeftIcon"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-chevron-right StyledProps--main StyledProps--padding-right-xsmall"
            icon="chevron-right"
          >
            <svg
              data-icon="chevron-right"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                chevron-right
              </desc>
              <path
                d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Next
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
